### DISCUSSION TITLE
Relationship between "tapToFocus" and "isAutoFocusEnabled"

### DISCUSSION THREAD

**Discussion by replicant1** [pedroSG94/RootEncoder#1521] [General]
Hi Pedro - I have a question about the "auto focus" feature in RootEncoder. When I call `RtmpCamera1.enableAutoFocus()`, if autofocus is supported by the camera, the Camera.Parameter `FOCUS_MODE_AUTO` is set and a field `autoFocusEnabled` is set to `true`. **But** when I call `tapToFocus`, **only** the `FOCUS_MODE_AUTO` parameter is set, and the field `autoFocusEnabled` is not changed. It looks to me like the field `autoFocusEnabled` ceases to be correlated with the camera's current focus mode once you call `tapToFocus`? Is this a bug - or perhaps a gap in my understanding? It seems to me that `autoFocusEnabled` should be set to `true` in `tapToFocus()` - either in the main body of the method or in the `autoFocusTakePictureCallback` referenced at the end of the method. Your thoughts?

**Comment by pedroSG94** [comment]
Hello,

Yes, this seem a bug. For now, you can call enableAutoFocus the first time before tapToFocus and this way you will solve the problem. Camera2 is affected too.
Also, I think that I should return a result in enableAutoFocus, disableAutoFocus and tapToFocus methods.
true if success, false if failed

**Comment by pedroSG94** [comment]
Fixed here:
https://github.com/pedroSG94/RootEncoder/commit/77bac3ca1e3c5ce0948fa71095ed69ca4b6cb954
Also, tapToFocus, enableAutoFocus and disableAutoFocus return a result (success true, fail false) to avoid need call isAutoFocusEnabled to know the result of the last call
